AK/LexicalPath.h

```diff
@@ -30,7 +30,8 @@ class LexicalPath {
 
     bool has_extension(StringView const&) const;
 
-    void append(String const& component);
+    [[nodiscard]] LexicalPath append(StringView const&) const;
+    [[nodiscard]] LexicalPath parent() const;
```

Comment:
This API came as a surprise (it's a nice API, but what's it gotta do with making `LexicalPath` immutable?)

Reply:
`void append` mutates the `LexicalPath` object, so I changed it to return a new `LexicalPath` object. The `parent()` somehow fell into the same commit since it's very related.

Reply:
Maybe mention it in the commit description, then?

Reply:
Ah, you did.
Nice.